flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/strategy/EagerSchedulingStrategyTest.java

```diff
@@ -0,0 +1,119 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.scheduler.strategy;
+
+import org.apache.flink.runtime.jobgraph.JobVertexID;
+import org.apache.flink.runtime.scheduler.ExecutionVertexDeploymentOption;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for {@link EagerSchedulingStrategy}.
+ */
+public class EagerSchedulingStrategyTest extends TestLogger {
+
+	private TestingSchedulerOperations testingSchedulerOperations;
+
+	private TestingSchedulingTopology testingSchedulingTopology;
+
+	private EagerSchedulingStrategy schedulingStrategy;
+
+	@Before
+	public void setUp() {
+		testingSchedulerOperations = new TestingSchedulerOperations();
+		testingSchedulingTopology = new TestingSchedulingTopology();
+		schedulingStrategy = new EagerSchedulingStrategy(
+				testingSchedulerOperations,
+				testingSchedulingTopology);
+	}
+	/**
+	 * Tests that when start scheduling eager scheduling strategy will start all vertices in scheduling topology.
+	 */
+	@Test
+	public void testStartScheduling() {
+		JobVertexID jobVertexID = new JobVertexID();
+		testingSchedulingTopology.addSchedulingExecutionVertex(new TestingSchedulingExecutionVertex(jobVertexID, 0));
+		testingSchedulingTopology.addSchedulingExecutionVertex(new TestingSchedulingExecutionVertex(jobVertexID, 1));
+		testingSchedulingTopology.addSchedulingExecutionVertex(new TestingSchedulingExecutionVertex(jobVertexID, 2));
+		testingSchedulingTopology.addSchedulingExecutionVertex(new TestingSchedulingExecutionVertex(jobVertexID, 3));
+		testingSchedulingTopology.addSchedulingExecutionVertex(new TestingSchedulingExecutionVertex(jobVertexID, 4));
+
+		schedulingStrategy.startScheduling();
+
+		assertThat(testingSchedulerOperations.getScheduledVertices(), hasSize(1));
+
+		Collection<ExecutionVertexDeploymentOption> scheduledVertices = testingSchedulerOperations.getScheduledVertices().get(0);
+		assertThat(scheduledVertices, hasSize(5));
+		for (SchedulingExecutionVertex schedulingExecutionVertex : testingSchedulingTopology.getVertices()) {
+			assertThat(getExecutionVertexIdsFromDeployOptions(scheduledVertices), hasItem(schedulingExecutionVertex.getId()));
```

Comment:
`getExecutionVertexIdsFromDeployOptions(scheduledVertices)` should be moved out of the loop.

Reply:
Done